data/modules/SearchRescue/SearchRescue.lua

```diff
@@ -2286,12 +2286,13 @@ end
 local onShipDestroyed = function (ship, attacker)
 	for _,mission in pairs(missions) do
 		if mission.target and ship == mission.target then
-			mission.target = nil
-			if attacker:IsPlayer() then
-				mission.target_destroyed = "BY_PLAYER"
-			else
-				mission.target_destroyed = "BY_ACCIDENT"
-			end
+		   mission.target = nil
+		   mission.target_destroyed = "BY_ACCIDENT"
+		   if attacker:IsDynamic() then
+		      if attacker:IsPlayer() then
+			 mission.target_destroyed = "BY_ACCIDENT"
+		      end
+		   end
```

Comment:
Looking good! I have only two things to suggest here:
- First of all, I'd recommend making sure to normalize your indentation, it looks like this is indented with three spaces instead of one tab as we'd prefer.
- Secondly, I'd recommend the use of idiomatic lua and write `if attacker:IsDynamic() and attacker:IsPlayer() then ... end`. Lua does *short-circuit* evaluation, so if the IsDynamic call fails the IsPlayer call will not execute.

And finally, did you mean for the target_destroyed reason to be `BY_PLAYER` instead of `BY_ACCIDENT` inside the if statement?

Reply:
@Web-eWorks - correct on all accounts. Thank you! I (hopefully) fixed the indentation, replaced BY_ACCIDENT with BY_PLAYER, and combined both if statements.